include/deal.II/base/array_view.h

```diff
@@ -286,6 +349,52 @@ make_array_view (std::vector<ElementType> &vector,
 }
 
 
+/**
+ * Create an ArrayView that takes a pair of iterators as arguments. The type
+ * of the ArrayView is inferred from the value type of the iterator (e.g., the
+ * view created from two const iterators will have a const type).
+ *
+ * @warning This function only makes sense for array or vector iterators
+ * (i.e., the memory between @p begin and @end contains valid objects). This
```

Comment:
I think you want to use the word "contiguous" here.

Is there a way to test whether an iterator range points to something contiguous? (Probably not, unfortunately.)